optuna/samplers/tpe/sampler.py

```diff
@@ -210,10 +210,14 @@ def _sample_discrete_uniform(self, distribution, below, above):
     def _sample_int(self, distribution, below, above):
         # type: (distributions.IntUniformDistribution, np.ndarray, np.ndarray) -> int
 
-        q = 1.0
-        low = distribution.low - 0.5 * q
-        high = distribution.high + 0.5 * q
-        return int(self._sample_numerical(low, high, below, above, q=q))
+        low = 0
+        high = int((distribution.high - distribution.low) / distribution.step)
```

Comment:
To be compatible with the previous logic, adding `+-0.5` to low and high values seems needed as follows:
```suggestion
        low = 0 - 0.5 
        high = int((distribution.high - distribution.low) / distribution.step) + 0.5
```

Reply:
BTW, the behavior of this method seems almost the same as `_sample_discrete_uniform`. How about invoking the method to reduce duplicate code as follows?
```python
    d = distributions.DiscreteUniformDistribution(
        low=distribution.low,
        high=distribution.high,
        q=distribution.step
    )
    return int(self._sample_discrete_uniform(d, below, above))
```

Reply:
> BTW, the behavior of this method seems almost the same as _sample_discrete_uniform

Looks good!